src/main/java/org/folio/fqm/lib/repository/IdStreamer.java

```diff
@@ -77,15 +77,17 @@ public int streamIdsInBatch(String tenantId,
   }
 
   public List<UUID> getSortedIds(String derivedTableName, EntityType entityType,
-                                 Condition sqlWhereClause, int offset, int batchSize) {
+                                 Condition sqlWhereClause, int offset, int batchSize, UUID queryId) {
+    // THIS DOES NOT PROVIDE SORTED IDs! This is a temporary workaround to address performance issues until we
+    // can do it properly
     return jooqContext.dsl()
-        .select(field(ID_FIELD_NAME))
-        .from(derivedTableName)
-        .where(sqlWhereClause)
-        .orderBy(getSortFields(entityType, true))
-        .offset(offset)
-        .limit(batchSize)
-        .fetchInto(UUID.class);
+      .select(field("result_id"))
+      .from(table(derivedTableName))
```

Comment:
I think the performance will be even better if we fetch from "query_results" table.